[RFC] log.c;message.c;strings.c add func_attr_printf

MichaHoffmann opened the pull request:
To add func_attr_printf to various functions and fix the resulting compiler warnings.

Commit:
add func_attr_printf in :

log.c
message.c
strings.c

fixed some printf warnings in:

src/nvim/undo.c
src/nvim/eval.c
src/nvim/eval/encode.c
src/nvim/eval/typval.c
src/nvim/ex_getln.c
src/nvim/fileio.c
src/nvim/lua/executor.c
src/nvim/main.c
src/nvim/regexp_nfa.c
src/nvim/shada.c
src/nvim/spellfile.c
src/nvim/tui/terminfo.c
src/nvim/garray.h

justinmk commented on src/nvim/buffer_updates.c:
```diff
@@ -180,7 +180,7 @@ void buf_updates_send_changes(buf_T *buf,
   // cleared up quickly.
   if (badchannelid != 0) {
-    ELOG("Disabling buffer updates for dead channel %llu", badchannelid);
+    ELOG("Disabling buffer updates for dead channel %lu", badchannelid);
```
I think we normally use `%zu` for `uint64_t` (and this is compatible with MSVC)

MichaHoffmann replied:
Okay, will change that!

jamessan replied:
Yeah, `%zu` would be incorrect on 32-bit platforms.  `"%" PRIu64` is the correct fix.

Commit:
src/nvim/eval.c: printf error

Commit:
fixed lint errors

MichaHoffmann commented on src/nvim/eval.c:
```diff
@@ -1940,10 +1940,10 @@ static char_u *ex_let_one(char_u *arg, typval_T *const tv,
     arg++;
     if (op != NULL && (*op == '+' || *op == '-')) {
-      emsgf(_(e_letwrong), op);
+      EMSG2(_(e_letwrong), op);
```
did not know that!

MichaHoffmann replied:
i'll leave the EMSG(msg) changes because emsg requires a char_u ptr.

Commit:
src/nvim/eval.c
src/nvim/eval/typval.c
src/nvim/fileio.c
src/nvim/main.c
src/nvim/regexp_nfa.c
src/nvim/spellfile.c

changed some EMSG[..] Macros to emsgf

justinmk commented:
Nice, thanks Michael!

We could remove `[I]EMSG{2,3}` entirely, they're useless.
